app/services/entitlement_service.py

```
@@ -47,27 +45,41 @@ def grant_hall_passes(
+
+    Args:
+        seat: Target seat receiving passes.
+        quantity: Number of passes to grant (must be positive).
+        actor_seat_id: Seat performing the action. Defaults to target seat.
+        trigger_id: Optional trigger identifier for payload lineage.
+        correlation_id: Cross-domain lineage ID. Generated if not provided.
+        acquisition_type: GRANT (teacher direct), PURCHASE, or PERK (rent).
+    """
     grant_quantity = int(quantity)
     if grant_quantity <= 0:
         raise ValueError("Hall-pass grant quantity must be positive")
```

Ruff (0.16.1)

[warning] Avoid specifying long messages outside the exception class

(TRY003)